src/server/scripts/EasternKingdoms/zone_silverpine_forest.cpp

```diff
@@ -1596,6 +1596,1436 @@ struct npc_silverpine_forsaken_bat : public VehicleAI
+                _events.ScheduleEvent(EVENT_ORC_PUP_TALK, 5s);
+
+                int c = player->GetReqKillOrCastCurrentCount(QUEST_STEEL_THUNDER, NPC_ORC_CRATE);
+
+                for (int i = 0; i < c; i++)
+                    DoCastSelf(SPELL_SUMMON_ORC_CRATE);
+            }
+        }
+    }
+
+    void PassengerBoarded(Unit* passenger, int8 seatId, bool apply) override
+    {
+        if (passenger->GetEntry() == NPC_ORC_CRATE)
+        {
+            if (apply)
+            {
+                if (_isJustSummoned)
+                    return;
+
+                if (Player* player = ObjectAccessor::GetPlayer(*me, _playerGUID))
+                    Talk(seatId + 1, player);
+            }
+            else
+                passenger->ToCreature()->DespawnOrUnsummon(3s);
+        }
+    }
+
+    void SpellHit(WorldObject* /*caster*/, SpellInfo const* spellInfo) override
+    {
+        switch (spellInfo->Id)
+        {
+            case SPELL_SEA_PUP_TRIGGER:
+                if (Player* player = ObjectAccessor::GetPlayer(*me, _playerGUID))
+                    Talk(TALK_ORC_PUP_DELIVER_CRATES, player);
+                if (Player* player = ObjectAccessor::GetPlayer(*me, _playerGUID))
+                    player->CastSpell(nullptr, SPELL_DESPAWN_ALL_SUMMONS, true);
+                _events.CancelEvent(EVENT_ORC_PUP_TALK);
+                break;
+
+            case SPELL_DESPAWN_ALL_SUMMONS:
+                _events.ScheduleEvent(EVENT_ORC_PUP_DELIVER_CRATES, 1s + 250ms);
+                break;
+            default:
+                break;
+        }
+    }
+
+    void UpdateAI(uint32 diff) override
+    {
+        if (Player* player = ObjectAccessor::GetPlayer(*me, _playerGUID))
+        {
+            if (!player->HasAura(SPELL_SUMMON_ORC_SEA_PUP))
+                me->DespawnOrUnsummon();
+        }
+
+        _events.Update(diff);
+
+        while (uint32 eventId = _events.ExecuteEvent())
+        {
+            switch (eventId)
+            {
+                case EVENT_ORC_PUP_JUST_SUMMONED:
+                    if (Player* player = ObjectAccessor::GetPlayer(*me, _playerGUID))
+                        Talk(TALK_ORC_PUP_SUMMONED, player);
+                    break;
+
+                case EVENT_ORC_PUP_TALK:
+                {
+                    if (me->GetVehicleKit()->IsVehicleInUse())
+                    {
+                        if (Player* player = ObjectAccessor::GetPlayer(*me, _playerGUID))
+                            Talk(TALK_ORC_PUP_WORN_OFF, player);
+                    }
+                    _events.ScheduleEvent(EVENT_ORC_PUP_TALK, 30s, 75s);
+                    break;
+                }
+
+                case EVENT_ORC_PUP_REMOVE_PROTECTION:
+                    _isJustSummoned = false;
+                    break;
+
+                case EVENT_ORC_PUP_DELIVER_CRATES:
+                    DoCastSelf(SPELL_EJECT_ALL_PASSENGERS);
+                    me->SetStandState(UNIT_STAND_STATE_DEAD);
```

Comment:
you might wanna remove spellclick flag here since the npc despawns 3s after player was ejected

Reply:
Feedback applied.